### PR TITLE
Slepc: Add E4S testsuite smoke test

### PR DESCRIPTION
**Pull request by RikkiButler20**
This PR represents a preliminary smoke test for the `slepc` package and is intended to serve the following purposes:

1. Leverage the current E4S test suite for the software
2. Demonstrate to package maintainers how to start a Spack smoke test.

### CODE REVIEW

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Please fix the flake8 errors:

```
var/spack/repos/builtin/packages/slepc/package.py:145: [E999] SyntaxError: invalid syntax
15
var/spack/repos/builtin/packages/slepc/package.py:146: [E112] expected an indented block
```

File: var/spack/repos/builtin/packages/slepc/test/hello.c [hunk 1]
```diff
@@ -0,0 +1,14 @@
+static char help[] = "Hello World example program\n";
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/slepc/test/hello.c:1]: Is this required?

Comment (RikkiButler20) [reply var/spack/repos/builtin/packages/slepc/test/hello.c:1]: I need it as an argument for `SlepcInitialize` so that I can use SLEPc classes.

**Review by tldahlgren — APPROVED** [review]
LGTM

**Pull request merged**